packages/react-components/src/hooks/useAlarms/transformers/parseAlarmStateProperty.ts

```diff
@@ -0,0 +1,44 @@
+import { AssetPropertyValue } from '@aws-sdk/client-iotsitewise';
+import { ALARM_STATUS } from '../constants';
+import { toTimestamp } from '../../../utils/time';
+
+export type UpperCaseStateName = keyof typeof ALARM_STATUS;
+export type PascalCaseStateName = (typeof ALARM_STATUS)[UpperCaseStateName];
+
+export const parseAlarmStateAssetProperty = (
+  assetPropertyValue?: AssetPropertyValue
+) => {
+  if (assetPropertyValue == null) {
+    return undefined;
+  }
+
+  const { timestamp, quality, value } = assetPropertyValue;
+
+  const alarmStateJSON = value?.stringValue;
+
+  if (alarmStateJSON == null) {
+    return undefined;
+  }
+
+  try {
+    const { stateName } = JSON.parse(alarmStateJSON);
+
+    let normalizedStateName: PascalCaseStateName;
+
+    if (ALARM_STATUS[stateName as UpperCaseStateName] != null) {
+      normalizedStateName = ALARM_STATUS[stateName as UpperCaseStateName];
+    } else {
+      normalizedStateName = stateName;
```

Comment:
Why are we setting an unknown state? We should probably log/throw an error

Reply:
I actually copied this from the existing alarms functionality in source-iotsitewise. I had assumed there was some legacy complication that we wanted to be able to display something atleast